volatility3/framework/plugins/windows/ldrmodules.py

```diff
@@ -0,0 +1,97 @@
+from volatility3.framework.symbols import intermed
+from volatility3.framework.symbols.windows.extensions import pe
+from volatility3.plugins.windows import pslist, vadinfo
+
+class LdrModules(interfaces.plugins.PluginInterface):
+    _required_framework_version = (2, 0, 0)
+    _version = (1, 0, 0)
+ 
+    @classmethod
+    def get_requirements(cls):
+        return [
+            requirements.ModuleRequirement(name = 'kernel', description = 'Windows kernel', architectures = ["Intel32", "Intel64"]),
+            requirements.SymbolTableRequirement(name = "nt_symbols", description = "Windows kernel symbols"),
+            requirements.VersionRequirement(name = 'pslist', component = pslist.PsList, version = (2, 0, 0)),
+            requirements.VersionRequirement(name = 'vadinfo', component = vadinfo.VadInfo, version = (2, 0, 0)),
+            requirements.ListRequirement(name = 'pid',
+                                         element_type = int,
+                                         description = "Process IDs to include (all other processes are excluded)",
+                                         optional = True),
+           ]
+
+    def _generator(self, procs):
+
+        pe_table_name = intermed.IntermediateSymbolTable.create(self.context,
+                                                                self.config_path,
+                                                                "windows",
+                                                                "pe",
+                                                                class_types = pe.class_types)
+
+        def filter_function(x: interfaces.objects.ObjectInterface) -> bool:
+            try:
+                return not (x.get_private_memory() == 0 and x.ControlArea)
+            except AttributeError:
+                return False
+
+        filter_func = filter_function   
+
+        for proc in procs:
+            proc_layer_name = proc.add_process_layer()
+
+            load_order_mod = dict((mod.DllBase, mod)
+                                for mod in proc.load_order_modules())
+            init_order_mod = dict((mod.DllBase, mod)
+                                for mod in proc.init_order_modules())
+            mem_order_mod = dict((mod.DllBase, mod)
+                                for mod in proc.mem_order_modules())
+
+            mapped_files = {}
+            for vad in vadinfo.VadInfo.list_vads(proc, filter_func = filter_func):
+                dos_header = self.context.object(pe_table_name + constants.BANG + "_IMAGE_DOS_HEADER",
+                                   offset = vad.get_start(),
+                                   layer_name = proc_layer_name)
+                try:
+                    if dos_header.e_magic != 0x5A4D:
+                        continue
+                except exceptions.PagedInvalidAddressException:
+                    continue
+
+                mapped_files[int(vad.get_start())] = str(vad.get_file_name() or '')
```

Comment:
I don't know that this needs to be cast as an int?  It should already be an int that gets returned, but if python says it's unhashable that would be good to know.  I'm just trying to avoid unnecessary casts (that might make other people include them too).  5:)  Same with the `str`.